ovsctl/ovsctl.go

```diff
@@ -55,7 +63,26 @@ type Ovsctl struct {
 }
 
 func NewOvsctl() Ovsctl {
-	return Ovsctl{execcli: platform.NewExecClient()}
+	execcli := platform.NewExecClient()
+	// try to ensure ovs-vswitchd is active (for Ubuntu 18), otherwise log failure to start the daemon
+	if output, err := execcli.ExecuteCommand("cat /etc/os-release"); err == nil {
+		if capture := osRE.FindStringSubmatch(output); len(capture) == 2 {
+			if fields := strings.Fields(capture[1]); fields[0] == "Ubuntu" {
```

Comment:
instead of doing nested ifs, how about you check for error and log it and return?